packages/contracts/src/contracts/current/protocol/Exchange/MixinWrapperFunctions.sol

```diff
@@ -327,7 +331,11 @@ contract MixinWrapperFunctions is
         for (uint256 i = 0; i < orders.length; i++) {
 
             // Token being sold by taker must be the same for each order
-            require(areBytesEqual(orders[i].takerAssetData, orders[0].takerAssetData));
+            // TODO: optimize by only using takerAssetData for first order.
```

Comment:
When will theres TODO's be addressed?

Reply:
I pretty much wrote this already in a different branch, but there are some tricky interactions with `fillOrderNoThrow`. I should be able to get to this next week.